lib/ansible/modules/cloud/amazon/ecr.py

```diff
@@ -0,0 +1,316 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import print_function
+
+import json
+import time
+import inspect
+
+from ansible.module_utils.basic import *
+from ansible.module_utils.ec2 import *
+
+try:
+    import boto3
+    from botocore.exceptions import ClientError
+
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ecr
+version_added: "2.3"
+short_description: Manage Elastic Container Registry repositories
+description:
+    - Manage Elastic Container Registry repositories
+options:
+    name:
+        description:
+            - the name of the repository
+        required: true
+    policy:
+        description:
+            - dict that gets JSONified for policy_text
+        required: false
+    policy_text:
+        description:
+            - string for new policy text
+            - setting to an empty string removes the repository's policy
+        required: false
+    delete_policy:
+        description:
+            - if yes, remove the policy from the repository
+        required: false
+        default: false
+    state:
+        description:
+            - create or destroy the repository
+        required: false
+        choices: [present, absent]
+        default: 'present'
+author:
+ - David M. Lee (@leedm777)
+extends_documentation_fragment: aws
+'''
+
+EXAMPLES = '''
+# Since neither policy nor policy_text are specified, existing policies on the
+# repository are unchanged
+- name: ecr-repo
+  ecr: name=super/cool
+
+- name: destroy-ecr-repo
+  ecr: name=old/busted state=absent
+
+- name: set-policy
+  ecr:
+    name: needs-policy
+    policy:
+      Version: '2008-10-17'
+      Statement:
+        - Sid: read-only
+          Effect: Allow
+          Principal:
+            AWS: '{{ read_only_arn }}'
+          Action:
+            - ecr:GetDownloadUrlForLayer
+            - ecr:BatchGetImage
+            - ecr:BatchCheckLayerAvailability
+
+- name: set-policy-text
+  ecr:
+    name: needs-policy
+    policy_text: "{{ lookup('template', 'policy.json.j2') }}"
```

Comment:
the thing is, if we remove the policy_text option, we can still do the following?

```
policy: "{{ lookup('template', 'policy.json.j2') | to_json }}"
```